src/java/org/apache/cassandra/cql3/WhereClause.java

```diff
@@ -105,35 +91,52 @@ public boolean equals(Object o)
             return false;
 
         WhereClause wc = (WhereClause) o;
-        return relations.equals(wc.relations) && expressions.equals(wc.expressions);
+        return expressionTree.equals(wc.expressionTree);
     }
 
     @Override
     public int hashCode()
     {
-        return Objects.hash(relations, expressions);
+        return Objects.hash(expressionTree);
     }
 
     public static final class Builder
```

Comment:
This is redundant with `ExpressionTree.Builder` why not use `ExpressionTree.Builder` directly?

Reply:
Agreed. I've merged the `ExpressionTree` into the `WhereClause` so `ExpressionTree` ceases to exist.